src/pages/Counter/containers/CounterContainer.js

```diff
@@ -0,0 +1,67 @@
+import React, { Component } from "react";
+import PropTypes from 'prop-types';
+
+import { CounterView } from "../components/CounterView/index.js";
+
+class CounterComponent extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      countValue: 0,
+      parityType: "even",
+    };
+  }
+
+  calculateParityType = (state) => {
+    return state.countValue % 2 === 0 ? "even" : "odd";
+  };
+
+  handleIncrement = () => {
+    this.setState((state) => {
+      return {
+        countValue: state.countValue + 1,
+      };
+    });
+    this.setState((state) => {
+      return {
+        parityType: this.calculateParityType(state),
+      };
+    });
+  };
+
+  handleDecrement = () => {
+    this.setState((state) => {
+      return {
+        countValue: state.countValue > 1 ? state.countValue - 1 : 0,
+      };
+    });
+    this.setState((state) => {
+      return {
+        parityType: this.calculateParityType(state),
+      };
```

Comment:
same as above. Also, check the `componentDidUpdate` lifecycle. Just imagine if you will have 10 handlers with keeping parity type logic. You will have to put this every time, but with `componentDidUpdate` you are able to write this once

Reply:
Move parity type to componentDidUpdate()